### DISCUSSION TITLE
build failed used macports,sb helper,thanks!

### DISCUSSION THREAD

**Discussion by laowupo** [LizardByte/community#371] [Sunshine] [answered]
*******Terminal:
Mac-mini:ports [myname]$ sudo port install sunshine -y
Password:
--->  Computing dependencies for Sunshine
--->  Building Sunshine
Error: Failed to build Sunshine: command execution failed
Error: See /opt/local/var/macports/logs/_Users_[myname]_ports_multimedia_sunshine/Sunshine/main.log for details.
Error: Follow https://guide.macports.org/#project.tickets if you believe there
is a bug.
Error: Processing of port sunshine failed

********main.log:
:debug:archivefetch Privilege de-escalation not attempted as not running as root.
:debug:archivefetch Skipping completed org.macports.configure (Sunshine)
:debug:archivefetch Privilege de-escalation not attempted as not running as root.
:debug:build build phase started at Mon Jan  8 09:07:47 CST 2024
:notice:build --->  Building Sunshine
:debug:build Executing proc-pre-org.macports.build-build-0
:debug:build system -W /opt/local/var/macports/build/_Users_[myname]_ports_multimedia_sunshine/Sunshine/work/Sunshine-0.21.0: npm install
:info:build npm ERR! code ERR_INVALID_URL
:info:build npm ERR! Invalid URL
:info:build npm ERR! A complete log of this run can be found in: /opt/local/var/macports/build/_Users_[myname]_ports_multimedia_sunshine/Sunshine/work/.home/.npm/_logs/2024-01-08T01_07_47_858Z-debug-0.log
:info:build Command failed: npm install
:info:build Exit code: 1
:error:build Failed to build Sunshine: command execution failed
:debug:build Error code: NONE
:debug:build Backtrace: command execution failed
:debug:build     while executing
:debug:build "$pre $targetname"

***********2024-01-08T01_07_47_858Z-debug-0.log:
42 timing idealTree:#root Completed in 16ms
43 timing idealTree:node_modules/@fortawesome/fontawesome-free Completed in 0ms
44 timing idealTree:node_modules/bootstrap Completed in 0ms
45 timing idealTree:node_modules/vue Completed in 0ms
46 timing idealTree:buildDeps Completed in 17ms
47 timing idealTree:fixDepFlags Completed in 0ms
48 timing idealTree Completed in 30ms
49 timing command:install Completed in 39ms
50 verbose stack TypeError [ERR_INVALID_URL]: Invalid URL
50 verbose stack     at new NodeError (node:internal/errors:405:5)
50 verbose stack     at new URL (node:internal/url:676:13)
50 verbose stack     at getProxyUri (/opt/local/lib/node_modules/npm/node_modules/make-fetch-happen/lib/agent.js:160:53)
50 verbose stack     at getAgent (/opt/local/lib/node_modules/npm/node_modules/make-fetch-happen/lib/agent.js:21:17)
50 verbose stack     at remoteFetch (/opt/local/lib/node_modules/npm/node_modules/make-fetch-happen/lib/remote.js:31:17)
50 verbose stack     at cacheFetch (/opt/local/lib/node_modules/npm/node_modules/make-fetch-happen/lib/cache/index.js:16:28)
50 verbose stack     at async fetch (/opt/local/lib/node_modules/npm/node_modules/make-fetch-happen/lib/fetch.js:98:7)
50 verbose stack     at async RegistryFetcher.packument (/opt/local/lib/node_modules/npm/node_modules/pacote/lib/registry.js:87:19)
50 verbose stack     at async RegistryFetcher.manifest (/opt/local/lib/node_modules/npm/node_modules/pacote/lib/registry.js:118:23)
50 verbose stack     at async #nodeFromEdge (/opt/local/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:1049:19)
51 verbose cwd /opt/local/var/macports/build/_Users_liao_ports_multimedia_sunshine/Sunshine/work/Sunshine-0.21.0
52 verbose Darwin 23.3.0
53 verbose node v18.19.0
54 verbose npm  v9.9.2
55 error code ERR_INVALID_URL
56 error Invalid URL
57 verbose exit 1
58 timing npm Completed in 284ms
59 verbose unfinished npm timer reify 1704676068102
60 verbose unfinished npm timer reify:loadTrees 1704676068108
61 verbose code 1

**Comment by julianrios** [comment]
I'm facing the same issue

**Reply by julianrios** [reply]
Seemed my issue was updating to version 0.22.0 reverting back to 0.21.0 fixed the build issue

**Reply by ReenigneArcher** [reply]
Try nightly version, I pushed some changes for macOS this morning.

**Reply by julianrios** [reply]
Dude you are fast! Thanks the Portfile seems to be missing some things to run the build I tried correcting on my end but didn't seem to work

**Reply by ReenigneArcher** [reply]
Please share the Portfile you have

**Reply by julianrios** [reply]
```
# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4

# initial PR into macports: https://github.com/macports/macports-ports/pull/15143

PortSystem               1.0
PortGroup                cmake 1.1
PortGroup                github 1.0
PortGroup                boost 1.0

name                     @PROJECT_NAME@
version                  @PROJECT_VERSION@
revision                 0
categories               multimedia emulators games
platforms                darwin
license                  GPL-3
maintainers              @LizardByte
description              @PROJECT_DESCRIPTION@

# long_description will not be split into multiple lines as it's configured by CMakeLists
long_description         @PROJECT_LONG_DESCRIPTION@
homepage                 @PROJECT_HOMEPAGE_URL@
master_sites             https://github.com/lizardbyte/sunshine/releases

compiler.cxx_standard    2017
fetch.type               git

git.url                  @GITHUB_CLONE_URL@
git.branch               @GITHUB_COMMIT@

post-fetch {
    system -W ${worksrcpath} "${git.cmd} submodule update --init --recursive"
}

depends_build-append     port:npm9 \
                         port:pkgconfig

depends_lib              port:avahi \
                         port:curl \
                         port:libopus \
                         port:miniupnpc

boost.version            1.81

configure.args           -DBUILD_WERROR=ON \
                         -DCMAKE_INSTALL_PREFIX=${prefix} \
                         -DSUNSHINE_ASSETS_DIR=etc/sunshine/assets

startupitem.create       yes
startupitem.executable   "${prefix}/bin/{$name}"
startupitem.location     LaunchDaemons
startupitem.name         ${name}
startupitem.netchange    yes

platform darwin {
    if { ${os.major} < 20 } {
        # See: https://github.com/LizardByte/Sunshine/discussions/117#discussioncomment-2513494
        notes-append "Port is limited to software encoding, when used with macOS releases prior to Big Sur."
    }
}

notes-append "Run @PROJECT_NAME@ by executing 'sunshine <path to user config>', e.g. 'sunshine ~/sunshine.conf' "
notes-append "The config file will be created if it doesn't exist."
notes-append "It is recommended to set a location for the apps file in the config."
notes-append "See our documentation at 'https://docs.lizardbyte.dev/projects/sunshine/en/v@PROJECT_VERSION@/' for further info."

```

**Reply by julianrios** [reply]
I tried updating as this:
```
# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4

# initial PR into macports: https://github.com/macports/macports-ports/pull/15143

PortSystem               1.0
PortGroup                cmake 1.1
PortGroup                github 1.0
PortGroup                boost 1.0

name                     Sunshine
version                  0.22.0
revision                 0
categories               multimedia emulators games
platforms                darwin
license                  GPL-3
maintainers              @LizardByte
description              Sunshine is a self-hosted game stream host for Moonlight.

# long_description will not be split into multiple lines as it's configured by CMakeLists
long_description         Offering low latency, cloud gaming server capabilities with support for AMD, Intel, and Nvidia GPUs for hardware encoding. Software encoding is also available. You can connect to Sunshine from any Moonlight client on a variety of devices. A web UI is provided to allow configuration, and client pairing, from your favorite web browser. Pair from the local server or any mobile device.
homepage                 https://app.lizardbyte.dev/Sunshine
master_sites             https://github.com/lizardbyte/sunshine/releases

compiler.cxx_standard    2017
fetch.type               git

git.url                  https://github.com/LizardByte/Sunshine.git
git.branch               894ef32c3b8208f6622b02c531fce14b33792c0c

post-fetch {
    system -W ${worksrcpath} "${git.cmd} submodule update --init --recursive"
}

depends_build-append     port:npm9 \
                         port:pkgconfig

depends_lib              port:avahi \
                         port:curl \
                         port:libopus \
                         port:miniupnpc

boost.version            1.81

configure.args           -DBUILD_WERROR=ON \
                         -DCMAKE_INSTALL_PREFIX=${prefix} \
                         -DSUNSHINE_ASSETS_DIR=etc/sunshine/assets

startupitem.create       yes
startupitem.executable   "${prefix}/bin/{$name}"
startupitem.location     LaunchDaemons
startupitem.name         ${name}
startupitem.netchange    yes

platform darwin {
    if { ${os.major} < 20 } {
        # See: https://github.com/LizardByte/Sunshine/discussions/117#discussioncomment-2513494
        notes-append "Port is limited to software encoding, when used with macOS releases prior to Big Sur."
    }
}

notes-append "Run @PROJECT_NAME@ by executing 'sunshine <path to user config>', e.g. 'sunshine ~/sunshine.conf' "
notes-append "The config file will be created if it doesn't exist."
notes-append "It is recommended to set a location for the apps file in the config."
notes-append "See our documentation at 'https://docs.lizardbyte.dev/projects/sunshine/en/v@PROJECT_VERSION@/' for further info."
```

**Reply by ReenigneArcher** [reply]
Get it from the "nightly" release. Link is on our site. https://app.lizardbyte.dev/Sunshine

**Reply by julianrios** [reply]
thanks grabbed from the nightly still seems to be facing some issue pulled this from the main.log
```
:info:build ✓ built in 955ms
:info:build make[2]: Leaving directory `/opt/local/var/macports/build/_Users_m2_ports_multimedia_sunshine/Sunshine/work/build'
:info:build [ 63%] Built target web-ui
:info:build make[1]: Leaving directory `/opt/local/var/macports/build/_Users_m2_ports_multimedia_sunshine/Sunshine/work/build'
:info:build make: *** [all] Error 2
:info:build make: Leaving directory `/opt/local/var/macports/build/_Users_m2_ports_multimedia_sunshine/Sunshine/work/build'
:info:build Command failed:  cd "/opt/local/var/macports/build/_Users_m2_ports_multimedia_sunshine/Sunshine/work/build" && /usr/bin/make -j12 -w all VERBOSE=ON 
:info:build Exit code: 2
:error:build Failed to build Sunshine: command execution failed
:debug:build Error code: CHILDSTATUS 9075 2
:debug:build Backtrace: command execution failed
:debug:build     while executing
:debug:build "system {*}$notty {*}$callback {*}$nice $fullcmdstring"
:debug:build     invoked from within
:debug:build "command_exec -callback portprogress::target_progress_callback build"
:debug:build     (procedure "portbuild::build_main" line 10)
:debug:build     invoked from within
:debug:build "$procedure $targetname"
```

**Reply by ReenigneArcher** [reply]
Post the full log

**Reply by julianrios** [reply]
[main.log](https://github.com/LizardByte/Sunshine/files/14501378/main.log)

**Reply by ReenigneArcher** [reply]
Okay, I discovered an issue in the CI... I will have it fixed shortly.

P.S. This is the actual error you're facing, which is what I have already fixed.
```txt
:info:build /opt/local/var/macports/build/_Users_julian_ports_multimedia_sunshine/Sunshine/work/Sunshine-0.22.0.9f94eeb/src/main.cpp:110:59: error: 'codecvt_utf8<wchar_t>' is deprecated [-Werror,-Wdeprecated-declarations]
:info:build   std::locale::global(std::locale(std::locale(), new std::codecvt_utf8<wchar_t>));
:info:build                                                           ^
:info:build /Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk/usr/include/c++/v1/codecvt:187:28: note: 'codecvt_utf8<wchar_t>' has been explicitly marked deprecated here
:info:build class _LIBCPP_TEMPLATE_VIS _LIBCPP_DEPRECATED_IN_CXX17 codecvt_utf8
:info:build                            ^
:info:build /Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk/usr/include/c++/v1/__config:862:41: note: expanded from macro '_LIBCPP_DEPRECATED_IN_CXX17'
:info:build #    define _LIBCPP_DEPRECATED_IN_CXX17 _LIBCPP_DEPRECATED
:info:build                                         ^
:info:build /Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk/usr/include/c++/v1/__config:835:49: note: expanded from macro '_LIBCPP_DEPRECATED'
:info:build #      define _LIBCPP_DEPRECATED __attribute__((deprecated))
:info:build                                                 ^
:info:build 1 error generated.
```

But now, the issue is the CI is not updating the Portfile in the release. I will keep up updated.

**Reply by ReenigneArcher** [reply, marked as answer]
nightly portfile is now fixed

**Reply by julianrios** [reply]
thanks! yea everything is working perfectly appreciate the quick help with this
